File: SPRING_BOOT/Modern_Hospital_Management_System/src/main/java/emranhss/com/Modern_Hospital_Management_System/service/EmergencyDoctorAssignmentService.java
package emranhss.com.Modern_Hospital_Management_System.service;

import emranhss.com.Modern_Hospital_Management_System.dto.request.EmergencyDoctorAssignmentRequest;
import emranhss.com.Modern_Hospital_Management_System.dto.response.EmergencyDoctorAssignmentResponse;

import java.util.List;

public interface EmergencyDoctorAssignmentService {

    EmergencyDoctorAssignmentResponse create(EmergencyDoctorAssignmentRequest request);

    List<EmergencyDoctorAssignmentResponse> getByEmergencyPatientId(Long emergencyPatientId);

    List<EmergencyDoctorAssignmentResponse> getAll();

    EmergencyDoctorAssignmentResponse unassign(Long id);

    List<EmergencyDoctorAssignmentResponse> getActiveAssignments();
}

File: SPRING_BOOT_Hospital_Management/Hospital_Management_system_angular/src/app/services/emergency/emergency-doctor-assignment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { EmergencyDoctorAssignment } from '../../models/emergency';

@Injectable({ providedIn: 'root' })
export class EmergencyDoctorAssignmentService {
  private api = environment.apiUrl + 'emergency/assignments';

  constructor(private http: HttpClient) {}

  create(assignment: EmergencyDoctorAssignment): Observable<EmergencyDoctorAssignment> {
    return this.http.post<EmergencyDoctorAssignment>(this.api, assignment);
  }

  getByEmergencyPatientId(emergencyPatientId: number): Observable<EmergencyDoctorAssignment[]> {
    return this.http.get<EmergencyDoctorAssignment[]>(`${this.api}/patient/${emergencyPatientId}`);
  }

  getAll(): Observable<EmergencyDoctorAssignment[]> {
    return this.http.get<EmergencyDoctorAssignment[]>(this.api);
  }

  unassign(id: number): Observable<void> {
    return this.http.put<void>(`${this.api}/${id}/unassign`, {});
  }

  getActiveAssignments(): Observable<EmergencyDoctorAssignment[]> {
    return this.http.get<EmergencyDoctorAssignment[]>(`${this.api}/active`);
  }
}
